routes/dashboard.rb

```diff
@@ -1,15 +1,34 @@
 class App < Sinatra::Base 
   get '/' do
     @cities = City.all
-    
-    @routes = Route.all.map do |route|
+
+    slim :index
+  end
+
+  post '/' do
+    if params[:departure_id].nil? || params[:departure_id].empty? || params[:arrival_id].nil? || params[:arrival_id].empty?
+      redirect '/'
+    else
+      query = request.params.map{|key, value| "#{key}=#{value}"}.join('&')
+      redirect "/search/#{query}"
+    end
+  end
+
+  get '/search/departure_id=:departure_id&arrival_id=:arrival_id?' do
+    @routes = Route.where({ start_id: params[:departure_id], destination_id: params[:arrival_id]}).map do |route|
       {
-        from: @cities.find { |c| c.id == route.start_id }.title,
-        departure_datetime: route.departure_datetime,
-        to: @cities.find { |c| c.id == route.destination_id }.title,
-        arrival_datetime: route.arrival_datetime,
+          departure_station: City.find_by_id(params[:departure_id]).title,
```

Comment:
Use `find` instead `find_by_id`. Your code will produce better for detecting error.
Or if you wrap error situation, use safe operator `&.title`
http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

Reply:
There's also `find_by` which doesn't produce an error but you'll have to handle `nil` case yourself

Reply:
But that way it needs to use `&.` as was mentioned

Reply:
Yeah, absolutely, just letting him know all of the alternatives